### DISCUSSION TITLE
In AR view, meshes vanish when camera is too close.

### DISCUSSION THREAD

**Discussion by FuzzyWobble** [google/model-viewer#2196] [Q&A]
When you move your camera too close to a model in AR some parts of the model mesh may disappear. This seems to be a common issue with more complex 3D models. This is usually fixed in threejs by setting frustumCulled to false.
[https://threejs.org/docs/#api/en/core/Object3D.frustumCulled](https://threejs.org/docs/#api/en/core/Object3D.frustumCulled)

Any idea how this can be fixed for model-viewer?

Thanks!

### Browser Affected

- [x ] Chrome

### OS

- [x ] Android

**Comment by elalish** [comment]
@FuzzyWobble If you're on Android in AR (and haven't flipped any flags in Chrome), then you're looking at SceneViewer, which is a separate app that doesn't use threejs at all. If you could give us a specific model to repro with (perhaps a Glitch?), I'll happily pass it along to that team, @tpsiaki. We will be supporting AR inside of Chrome shortly via webXR, so that should also fix this issue.

**Comment by FuzzyWobble** [comment]
Oh  :(  I see. 
I didn't realize scene-viewer and model-viewer were completely separate. 
Thanks for your reply. 

I can't share this exact model right now but let me try and find one that also has this bug and share a glitch link.

**Comment by FuzzyWobble** [comment]
@elalish Hello. I can't find a git repo for scene-viewer or anywhere to post issues. Is it possible you can put me in touch with someone from there? I have a number of issues and strange behaviors. Thanks!

**Comment by elalish** [comment]
@FuzzyWobble Yes, we commonly take scene-viewer issues here for that reason. Feel free to continue the discussion and I'll loop in @tpsiaki from the scene-viewer team.

**Comment by FuzzyWobble** [comment]
Oh thanks so much!

1. Meshes are disappearing on AR scene-veiwer when the camera is close to model. Typically this is solved with [frustumculled](https://discourse.threejs.org/t/zooming-in-camera-make-some-meshes-not-visible/3872/8) in threejs. How would this be solved for scene-viewer?

2. Using the fallback url approach I cannot get any gltf AR to work on iOS-Chrome as documented [here](https://developers.google.com/ar/develop/java/scene-viewer). Is it possible?

**Comment by FuzzyWobble** [comment]
*bumpity bumpity*

**Comment by tpsiaki** [comment]
Hi @FuzzyWobble - The issues you describe seem to indicate that the bounds in your glTF don't match the content (bounds are specified in the "accessors" list as "min"/"max" for each accessor). This would explain why the model would disappear under certain circumstances in Scene Viewer, and require frustumculled in threejs.

Are you exporting these gltfs with a particular pipeline that might not be calculating bounds correctly?

**Comment by elalish** [comment]
@FuzzyWobble As for question 2, on iOS we launch QuickLook instead of SceneViewer (which is Android), so those directions do not apply.

**Comment by FuzzyWobble** [comment]
Ah nice. Got it. Thanks so much!
